dotnet/samples/Concepts/RAG/Bing_RagWithTextSearch.cs

```diff
@@ -0,0 +1,186 @@
+                modelId: TestConfiguration.OpenAI.ChatModelId,
+                apiKey: TestConfiguration.OpenAI.ApiKey);
+        Kernel kernel = kernelBuilder.Build();
+
+        // Create a text search using Bing search
+        var textSearch = new BingTextSearch(new(TestConfiguration.Bing.ApiKey));
+
+        // Build a text search plugin with Bing search and add to the kernel
+        var searchPlugin = textSearch.CreateWithGetTextSearchResults("SearchPlugin");
+        kernel.Plugins.Add(searchPlugin);
+
+        // Invoke prompt and use text search plugin to provide grounding information
+        var query = "What is the Semantic Kernel?";
+        string promptTemplate = @"
+{{#with (SearchPlugin-GetTextSearchResults query)}}  
+  {{#each this}}  
+    Name: {{Name}}
+    Value: {{Value}}
+    Link: {{Link}}
+    -----------------
+  {{/each}}  
+{{/with}}  
+
+{{query}}
+
+Include citations to the relevant information where it is referenced in the response.
+";
+        KernelArguments arguments = new() { { "query", query } };
+        HandlebarsPromptTemplateFactory promptTemplateFactory = new();
+        Console.WriteLine(await kernel.InvokePromptAsync(
+            promptTemplate,
+            arguments,
+            templateFormat: HandlebarsPromptTemplateFactory.HandlebarsTemplateFormat,
+            promptTemplateFactory: promptTemplateFactory
+        ));
+    }
+
+    /// <summary>
+    /// Show how to create a default <see cref="KernelPlugin"/> from an <see cref="ITextSearch"/> and use it to
+    /// add grounding context to a Handlebars prompt and include citations in the response.
+    /// </summary>
+    [Fact]
+    public async Task RagWithBingTextSearchIncludingTimeStampedCitationsAsync()
+    {
+        // Create a kernel with OpenAI chat completion
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.AddOpenAIChatCompletion(
+                modelId: TestConfiguration.OpenAI.ChatModelId,
+                apiKey: TestConfiguration.OpenAI.ApiKey);
+        Kernel kernel = kernelBuilder.Build();
+
+        // Create a text search using Bing search
+        var textSearch = new BingTextSearch(new(TestConfiguration.Bing.ApiKey));
+
+        // Build a text search plugin with Bing search and add to the kernel
+        var searchPlugin = textSearch.CreateWithGetSearchResults("SearchPlugin");
+        kernel.Plugins.Add(searchPlugin);
+
+        // Invoke prompt and use text search plugin to provide grounding information
+        var query = "What is the Semantic Kernel?";
+        string promptTemplate = @"
+{{#with (SearchPlugin-GetSearchResults query)}}  
+  {{#each this}}  
+    Name: {{Name}}
+    Snippet: {{Snippet}}
+    Link: {{DisplayUrl}}
+    Date Last Crawled: {{DateLastCrawled}}
+    -----------------
+  {{/each}}  
+{{/with}}  
+
+{{query}}
+
+Include citations to and the date of the relevant information where it is referenced in the response.
+";
```

Comment:
nit. Suggest using the [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/raw-string-literal#summary) as they remove the indentation automatically. `"""` 


i.e:
```suggestion
        string promptTemplate = """
                                {{#with (SearchPlugin-GetSearchResults query)}}  
                                  {{#each this}}  
                                    Name: {{Name}}
                                    Snippet: {{Snippet}}
                                    Link: {{DisplayUrl}}
                                    Date Last Crawled: {{DateLastCrawled}}
                                    -----------------
                                  {{/each}}  
                                {{/with}}  
                                
                                {{query}}
                                
                                Include citations to and the date of the relevant information where it is referenced in the response.
                                """;
```

Reply:
Updated all of the strings in the samples